### PR TITLE
Add Exclusion Frontmatter for TFC-Only Docs

### PR DESCRIPTION
**Pull request by laurapacilio**
### What
Web Development is awesome and made it so we can add frontmatter to the TFC Only pages we want to exclude from the Terraform Enterprise versioned documentation set. This PR adds frontmatter to the current list of excluded pages. Specifically, we're excluding:

- [Recommended Practices](https://github.com/hashicorp/terraform-docs-common/tree/main/website/docs/cloud-docs/recommended-practices)
- [What is Terraform Cloud?](https://github.com/hashicorp/terraform-docs-common/blob/main/website/docs/cloud-docs/index.mdx)
- [Plans and Features](https://github.com/hashicorp/terraform-docs-common/blob/main/website/docs/cloud-docs/overview.mdx)
- [Architectural Details - Overview](https://github.com/hashicorp/terraform-docs-common/blob/main/website/docs/cloud-docs/architectural-details/index.mdx)
- [Architecture Details - IP Ranges](https://github.com/hashicorp/terraform-docs-common/blob/main/website/docs/cloud-docs/architectural-details/ip-ranges.mdx)
- [Architectural Details - Security Model](https://github.com/hashicorp/terraform-docs-common/blob/main/website/docs/cloud-docs/architectural-details/security-model.mdx)
- [Integrations - Splunk](https://github.com/hashicorp/terraform-docs-common/tree/main/website/docs/cloud-docs/integrations/splunk)
- [Integrations - Kubernetes](https://github.com/hashicorp/terraform-docs-common/tree/main/website/docs/cloud-docs/integrations/kubernetes)
- [GitHub.com VCS Instructions](https://github.com/hashicorp/terraform-docs-common/blob/main/website/docs/cloud-docs/vcs/github-app.mdx)
- [Drift Detection](https://github.com/hashicorp/terraform-docs-common/blob/main/website/docs/cloud-docs/workspaces/settings/drift-detection.mdx)
- [SSO - all pages](https://github.com/hashicorp/terraform-docs-common/tree/main/website/docs/cloud-docs/users-teams-organizations/single-sign-on)
- [IP Ranges API Documentation](https://github.com/hashicorp/terraform-docs-common/blob/main/website/docs/cloud-docs/api-docs/ip-ranges.mdx)

### Why
TFE users shouldn't have to see content that's not relevant to them. This PR helps us exclude TFC-Specific information from the TFE docs. It also reduces the manual work involved with prepping TFE versioned docs. Before this, we had to manually edit the TFE docs PR to exclude these files.

----------

### Merge Checklist
_If items do not apply to your changes, add (N/A) and mark them as complete._

#### Pull Request
- [x] One or more labels describe the type of change (e.g. clarification) and associated product (e.g. tfc).
- [x] Description links to related pull requests or issues, if any.

#### Content
- [x] Redirects have been added for moved, renamed, or deleted pages. This requires a separate PR in the [`terraform-website` repository](https://github.com/hashicorp/terraform-website) `redirects.next.js` file.
- [x] API documentation and the API Changelog have been updated. 
- [x] Links to related content where appropriate (e.g., API endpoints, permissions, etc.).
- [x] Pages with related content are updated and link to this content when appropriate.
- [x] Sidebar navigation files have been updated for added, deleted, reordered, or renamed pages.
- [x] New pages have metadata (page name and description) at the top.
- [x] New images are 2048 px wide. They have HashiCorp standard annotation color (#F92672) and format (rectangle with rounded corners), blurred sensitive details (e.g. credentials, usernames, user icons), and descriptive alt text in the markdown for accessibility.
- [x] New code blocks have the correct syntax and line breaks to eliminate horizontal scroll bars.
- [x] UI elements (button names, page names, etc.) are bolded.
- [x] The Vercel website preview successfully deployed.

#### Reviews
- [x] I or someone else reviewed the content for technical accuracy.
- [x] I or someone else reviewed the content for typos, punctuation, spelling, and grammar.

### CODE REVIEW

File: website/docs/cloud-docs/vcs/github-app.mdx [hunk 1]
```diff
@@ -18,7 +19,7 @@ These instructions are for using repositories from GitHub.com with Terraform Clo
 
 This method uses a preconfigured GitHub App, and only works with GitHub.com. There are separate instructions for connecting to [GitHub.com via OAuth](/cloud-docs/vcs/github), connecting to [GitHub Enterprise](/cloud-docs/vcs/github-enterprise), and connecting to [other supported VCS providers.](/cloud-docs/vcs)
 
--> **Note:** This VCS Provider is only available on Terraform Cloud. If you are using Terraform Enterprise, you will need to follow the instructions for connecting to [GitHub.com via OAuth](/cloud-docs/vcs/github)
+-> **Note:** This VCS Provider is only available on Terraform Cloud. If you are using Terraform Enterprise, follow the instructions for connecting to [GitHub.com via OAuth](/cloud-docs/vcs/github)
```
Comment (laurapacilio) [inline website/docs/cloud-docs/vcs/github-app.mdx:22]:
```suggestion
-> **Note:** This VCS Provider is only available on Terraform Cloud. If you are using Terraform Enterprise, follow the instructions for connecting to [GitHub.com via OAuth](/cloud-docs/vcs/github).
```

File: website/docs/cloud-docs/recommended-practices/part3.mdx [hunk 2]
```diff
@@ -2,6 +2,7 @@
 page_title: >-
   Part 3: How to Evolve Your Provisioning Practices - Terraform Recommended
   Practices
+  tfc_only: true
```
Comment (thiskevinwang) [inline website/docs/cloud-docs/recommended-practices/part3.mdx:5]:
```suggestion
tfc_only: true
```

File: website/docs/cloud-docs/recommended-practices/part2.mdx [hunk 3]
```diff
@@ -2,6 +2,7 @@
 page_title: >-
   Part 2: Evaluating Your Current Provisioning Practices - Terraform Recommended
   Practices
+  tfc_only: true
```
Comment (thiskevinwang) [inline website/docs/cloud-docs/recommended-practices/part2.mdx:5]:
```suggestion
tfc_only: true
```

File: website/docs/cloud-docs/recommended-practices/part3.1.mdx [hunk 4]
```diff
@@ -2,6 +2,7 @@
 page_title: >-
   Part 3.1: From Manual Changes to Semi-Automation - Terraform Recommended
   Practices
+  tfc_only: true
```
Comment (thiskevinwang) [inline website/docs/cloud-docs/recommended-practices/part3.1.mdx:5]:
```suggestion
tfc_only: true
```

File: website/docs/cloud-docs/recommended-practices/part3.2.mdx [hunk 5]
```diff
@@ -2,6 +2,7 @@
 page_title: >-
   Part 3.2: From Semi-Automation to Infrastructure as Code - Terraform
   Recommended Practices
+  tfc_only: true
```
Comment (thiskevinwang) [inline website/docs/cloud-docs/recommended-practices/part3.2.mdx:5]:
```suggestion
tfc_only: true
```

File: website/docs/cloud-docs/recommended-practices/part3.3.mdx [hunk 6]
```diff
@@ -2,6 +2,7 @@
 page_title: >-
   Part 3.3: From Infrastructure as Code to Collaborative Infrastructure as Code
   - Terraform Recommended Practices
+  tfc_only: true
```
Comment (thiskevinwang) [inline website/docs/cloud-docs/recommended-practices/part3.3.mdx:5]:
```suggestion
tfc_only: true
```

**Review by thiskevinwang — CHANGES_REQUESTED** [review]
Let a few indentation fixes as suggestions. Otherwise, looks good!

**Review by thiskevinwang — APPROVED** [review]

**Pull request merged**